src/components/ProductList.tsx

```diff
@@ -76,46 +139,52 @@ function ProductList() {
 
   return (
     <>
-      <List>
-        {visibleProducts.map((p) => (
-          <Item
-            key={p.id}
-            onClick={() => handleItemClick(p.id)}
-            style={{ cursor: 'pointer' }}
-          >
-            {/* 상품 이미지 */}
-            <img
-              src={p.imageURL}
-              alt={p.name}
-              style={{ width: '100%', borderRadius: 8 }}
-            />
-            {/* 브랜드명 (회색, 작은 글씨) */}
-            <div style={{ color: '#888', fontSize: 14, marginTop: 8 }}>
-              {p.brandInfo.name}
-            </div>
-            {/* 상품명 (굵은 글씨) */}
-            <div style={{ fontWeight: 700, fontSize: 16, marginTop: 4 }}>
-              {p.name}
-            </div>
-            {/* 가격 (굵고 큰 글씨) */}
-            <div style={{ fontWeight: 700, fontSize: 18, marginTop: 4 }}>
-              {p.price.sellingPrice.toLocaleString()} 원
-            </div>
-          </Item>
-        ))}
-      </List>
-      {/* 더보기/접기 버튼 */}
-      {filtered.length > DEFAULT_VISIBLE && (
+      {loading ? (
+        <div>로딩 중 ...</div>
+      ) : error ? (
+        <div>{error}</div>
+      ) : visibleProducts.length === 0 ? (
+        <div>상품 목록이 없습니다.</div>
+      ) : (
         <>
-          {!isAllVisible && (
-            <MoreButton onClick={() => setVisibleCount(filtered.length)}>
-              더보기
-            </MoreButton>
-          )}
-          {isAllVisible && (
-            <MoreButton onClick={() => setVisibleCount(DEFAULT_VISIBLE)}>
-              접기
-            </MoreButton>
+          <List>
+            {visibleProducts.map((p, idx) => (
+              <Item
+                key={p.id}
+                onClick={() => handleItemClick(p.id)}
+                style={{ cursor: 'pointer' }}
+              >
+                <RankBadge rank={idx + 1}>{idx + 1}</RankBadge>
+                <img
+                  src={p.imageURL}
+                  alt={p.name}
+                  style={{ width: '100%', borderRadius: 8 }}
+                />
+                <div style={{ color: '#888', fontSize: 14, marginTop: 8 }}>
+                  {p.brandInfo.name}
+                </div>
+                <div style={{ fontWeight: 700, fontSize: 16, marginTop: 4 }}>
+                  {p.name}
+                </div>
+                <div style={{ fontWeight: 700, fontSize: 18, marginTop: 4 }}>
+                  {p.price.sellingPrice.toLocaleString()} 원
+                </div>
+              </Item>
+            ))}
+          </List>
+          {products.length > DEFAULT_VISIBLE && (
+            <>
+              {!isAllVisible && (
+                <MoreButton onClick={() => setVisibleCount(products.length)}>
+                  더보기
+                </MoreButton>
+              )}
+              {isAllVisible && (
+                <MoreButton onClick={() => setVisibleCount(DEFAULT_VISIBLE)}>
+                  접기
+                </MoreButton>
+              )}
```

Comment:
여기 지금봤는데!...
3항 연산자로 처리하는게 더 깔끔해보여요
```suggestion
              {!isAllVisible ? (
                <MoreButton onClick={() => setVisibleCount(products.length)}>
                  더보기
                </MoreButton>
              ) : 
                <MoreButton onClick={() => setVisibleCount(DEFAULT_VISIBLE)}>
                  접기
                </MoreButton>
              }
```